src/stactools/viirs/cog.py

```diff
@@ -133,7 +135,7 @@ def _cog(
         with mem_file.open(**src_profile) as mem:
             mem.write(data, 1)
             mem.update_tags(**tags)
-            stactools.core.utils.convert.cogify(mem, cog_path)
+            rasterio.shutil.copy(mem, cog_path, **COG_PROFILE)
```

Comment:
I tried to crib this function (or one like it) for https://github.com/stac-utils/stactools/blob/2da69d8bea9d31a82bdcb4b48472d825aa112434/src/stactools/core/utils/convert.py#L78-L120. For sure not required to use here, just curious if it would suit your purposes.

Reply:
Yeah, I might be able to go back and adapt things to use the new stuff in stactools. It would be a bit of refactoring, though. I think there would still be some odd hoops to jump through with the check for signed int8 data and the multiple nodata value cleaning.